src/codemagic/tools/app_store_connect.py

```diff
@@ -492,25 +492,41 @@ def list_devices(
         "register-device",
         BundleIdArgument.PLATFORM,
         DeviceArgument.DEVICE_NAME,
-        DeviceArgument.DEVICE_UDID,
+        DeviceArgument.DEVICE_UDIDS,
+        DeviceArgument.IGNORE_REGISTRATION_ERRORS,
     )
     def register_device(
         self,
         platform: BundleIdPlatform,
         device_name: str,
-        device_udid: str,
+        device_udids: Types.DeviceUdidsArgument,
+        ignore_registration_errors: bool = False,
         should_print: bool = True,
-    ) -> Device:
+    ) -> List[Device]:
         """
-        Register a new device for app development
+        Register new Devices for app development
         """
-        return self._create_resource(
-            self.api_client.devices,
-            should_print,
-            udid=device_udid,
-            name=device_name,
-            platform=platform,
-        )
+        registered_devices = []
+        for i, device_udid in enumerate(Types.DeviceUdidsArgument.resolve_value(device_udids)):
+            if should_print:
+                self.echo("") if i != 0 else None
+
+            try:
+                device = self._create_resource(
+                    self.api_client.devices,
+                    should_print,
+                    udid=device_udid,
+                    name=device_name,
+                    platform=platform,
+                )
+            except AppStoreConnectError as error:
+                if not ignore_registration_errors:
+                    raise error from error
```

Comment:
Use just `raise` as doing `raise error from error` causes unnecessary traceback stacking.

<details>
<summary>Example</summary>

```python
try:
    1/0
except ZeroDivisionError as zde:
    print("Oopsie, division by zero")
    raise zde from zde
```
runs as
```python
Oopsie, division by zero
Traceback (most recent call last):
  File "/private/tmp/error.py", line 5, in <module>
    raise zde from zde
  File "/private/tmp/error.py", line 2, in <module>
    1/0
    ~^~
ZeroDivisionError: division by zero
```
whereas
```python
try:
    1/0
except ZeroDivisionError as zde:
    print("Oopsie, division by zero")
    raise
```
points directly to the original cause
```python
Oopsie, division by zero
Traceback (most recent call last):
  File "/private/tmp/error.py", line 2, in <module>
    1/0
    ~^~
ZeroDivisionError: division by zero
```

</details>